### PR TITLE
Fix sending info message for ResetGlobalProperties after timeout

### PR DESCRIPTION
**Pull request by VVeremjova**
Fixed sending HMI response of UI.SetGlobalProperties, TTS.SetGlobalProperties or both.

Add unit tests for check timeout for ResetGlobalProperties

### CODE REVIEW

**Comment by VVeremjova** [comment]
@LevchenkoS, @pold500 please review

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 1]
```diff
@@ -330,6 +320,185 @@ TEST_F(ResetGlobalPropertiesResponseTest, Run_Sendmsg_SUCCESS) {
...
+  EXPECT_CALL(
+      app_mngr_,
+      ManageMobileCommand(
+          MobileResponseIs(mobile_apis::Result::GENERIC_ERROR, info, false),
+          am::commands::Command::ORIGIN_SDL));
+  command_->onTimeOut();
+}
+
+TEST_F(ResetGlobalPropertiesRequestTest,
+       Run_WaitUIAndTTS_Timeout_GENERIC_ERROR_TTSUINotRespond) {
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @VVeremjova I would suggest to rename this test into:
Run_WaitUIAndTTS_Timeout_GENERIC_ERROR_UIAndTTSNotRespond

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 2]
```diff
@@ -330,6 +320,185 @@ TEST_F(ResetGlobalPropertiesResponseTest, Run_Sendmsg_SUCCESS) {
...
+  (*ui_msg)[am::strings::params][am::strings::correlation_id] = kCorrelationId;
+  (*ui_msg)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::eType::SUCCESS;
+
+  Event ui_event(hmi_apis::FunctionID::UI_SetGlobalProperties);
+  ui_event.set_smart_object(*ui_msg);
+
+  command_->on_event(ui_event);
+
+  // TTS doesn't respond, so timeout should send generic error
```
Comment (dcherniev) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @VVeremjova This comment can be quite handy, I think the other two new tests should also have such comment. Also I would suggest to delete empty line below this comment.

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 3]
```diff
@@ -330,6 +320,185 @@ TEST_F(ResetGlobalPropertiesResponseTest, Run_Sendmsg_SUCCESS) {
...
+  EXPECT_CALL(app_mngr_settings_, time_out_promt())
+      .WillOnce(ReturnRef(time_out_prompt));
+
+  EXPECT_CALL(*mock_app_, set_timeout_prompt(_));
+
+  smart_objects::SmartObjectSPtr prompt =
+      utils::MakeShared<smart_objects::SmartObject>();
+  *prompt = "prompt";
+
+  EXPECT_CALL(*mock_app_, timeout_prompt()).WillOnce(Return(prompt.get()));
```
Comment (tkireva) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @VVeremjova Maybe you mean prompt->get()? (and all same line below)

Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @tkireva No, prompt.get() returns raw pointer `smart_objects::SmartObject*` , which actually needs here

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 1 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev See 5815908

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc (hunk 2 above)
Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @dcherniev See 5815908

File: src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc [hunk 4]
```diff
@@ -330,6 +320,183 @@ TEST_F(ResetGlobalPropertiesResponseTest, Run_Sendmsg_SUCCESS) {
...
+      mobile_apis::GlobalProperty::TIMEOUTPROMPT;
+  (*msg_)[am::strings::msg_params][am::strings::properties][1] =
+      mobile_apis::GlobalProperty::MENUICON;
+
+  std::vector<std::string> time_out_prompt;
+  time_out_prompt.push_back("time_out");
+  EXPECT_CALL(app_mngr_settings_, time_out_promt())
+      .WillOnce(ReturnRef(time_out_prompt));
+
+  EXPECT_CALL(*mock_app_, set_timeout_prompt(_));
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @VVeremjova, are we really don't interested in argument value?

Comment (VVeremjova) [inline src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc]: @LevchenkoS Yes. We only need to be sure that something was send to TTS, which  will be true if set_timeout_prompt was called

**Comment by VVeremjova** [comment]
Rebuild required

**Comment by VVeremjova** [comment]
Rebuild required

**Comment by VVeremjova** [comment]
Rebuild required

**Pull request merged**